src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+        $scriptId = $this->getScriptId($httpClient, 'script_not_client_pre');
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withPreScripts($clientName, [$scriptId]);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('Script "'.$scriptId.'" is not a client pre script');
+    }
+    public function testCreateClientUnexistentPostScript(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withPostScripts($clientName, [self::UNEXISTING_ID]);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('Script "'.self::UNEXISTING_ID.'" does not exist');
+    }
+
+    public function testCreateClientUnexistentPreScript(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withPreScripts($clientName, [self::UNEXISTING_ID]);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('Script "'.self::UNEXISTING_ID.'" does not exist');
+    }
+
+    public function testDeleteClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client_to_delete'
+        ]);
+        $response = $httpClient->request('DELETE', $iri);
+        $this->assertResponseIsSuccessful();
+        $response = $httpClient->request('GET', $iri);
+        $this->assertResponseStatusCodeSame(404);
+    }
+
+    public function testDeleteClientNotFound(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('DELETE', 'api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClients(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
```

Comment:
Could we at least test for a well known client so that we make sure that the output format is the expected one?